wpiutil/src/main/native/include/wpi/jni_util.h

```diff
@@ -175,197 +175,276 @@ class JStringRef {
-      }                                                                        \
-    }                                                                          \
-  };                                                                           \
-                                                                               \
-  class CriticalJ##F##ArrayRef : public detail::JArrayRefBase<T> {             \
-   public:                                                                     \
-    CriticalJ##F##ArrayRef(JNIEnv* env, T##Array jarr, int len)                \
-        : detail::JArrayRefBase<T>(env, jarr, len) {                           \
-      if (jarr) {                                                              \
-        m_elements =                                                           \
-            static_cast<T*>(env->GetPrimitiveArrayCritical(jarr, nullptr));    \
-      } else {                                                                 \
-        errs() << "JArrayRef was passed a null pointer at \n"                  \
-               << GetJavaStackTrace(env);                                      \
-      }                                                                        \
-    }                                                                          \
-    CriticalJ##F##ArrayRef(JNIEnv* env, T##Array jarr)                         \
-        : detail::JArrayRefBase<T>(env, jarr) {                                \
-      if (jarr) {                                                              \
-        m_elements =                                                           \
-            static_cast<T*>(env->GetPrimitiveArrayCritical(jarr, nullptr));    \
-      } else {                                                                 \
-        errs() << "JArrayRef was passed a null pointer at \n"                  \
-               << GetJavaStackTrace(env);                                      \
-      }                                                                        \
-    }                                                                          \
-    ~CriticalJ##F##ArrayRef() {                                                \
-      if (m_jarr && m_elements) {                                              \
-        m_env->ReleasePrimitiveArrayCritical(m_jarr, m_elements, JNI_ABORT);   \
-      }                                                                        \
-    }                                                                          \
-  };
+template <typename T, size_t Extent = std::dynamic_extent>
+using JSpan = detail::JSpanBase<T, false, Extent>;
+
+template <typename T, size_t Extent = std::dynamic_extent>
+using CriticalJSpan = detail::JSpanBase<T, true, Extent>;
+
+#define WPI_JNI_JARRAYREF(T, F)          \
+  using J##F##ArrayRef = JSpan<const T>; \
+  using CriticalJ##F##ArrayRef = CriticalJSpan<const T>;
```

Comment:
I think we don't need the shims--this header isn't team-facing.  We should just fix up our uses.